homeassistant/components/twinkly/light.py

```diff
@@ -148,31 +148,50 @@ async def async_turn_on(self, **kwargs) -> None:
             # If brightness is 0, the twinkly will only "disable" the brightness,
             # which means that it will be 100%.
             if brightness == 0:
-                await self._client.set_is_on(False)
+                await self._client.turn_off()
                 return
 
             await self._client.set_brightness(brightness)
 
-        await self._client.set_is_on(True)
+        if ATTR_RGBW_COLOR in kwargs:
+            if kwargs[ATTR_RGBW_COLOR] != self._attr_rgbw_color:
+                self._attr_rgbw_color = kwargs[ATTR_RGBW_COLOR]
+
+                if isinstance(self._attr_rgbw_color, tuple):
+
+                    await self._client.interview()
+                    # Reagarrange from rgbw to wrgb
+                    await self._client.set_static_colour(
+                        (
+                            self._attr_rgbw_color[3],
+                            self._attr_rgbw_color[0],
+                            self._attr_rgbw_color[1],
+                            self._attr_rgbw_color[2],
+                        )
+                    )
+
+        if not self._is_on:
+            await self._client.turn_on()
 
     async def async_turn_off(self, **kwargs) -> None:
         """Turn device off."""
-        await self._client.set_is_on(False)
+        await self._client.turn_off()
 
     async def async_update(self) -> None:
         """Asynchronously updates the device properties."""
         _LOGGER.info("Updating '%s'", self._client.host)
 
         try:
-            self._is_on = await self._client.get_is_on()
+            self._is_on = await self._client.is_on()
 
-            self._brightness = (
-                int(round((await self._client.get_brightness()) * 2.55))
-                if self._is_on
-                else 0
+            brightness = await self._client.get_brightness()
+            brightness_value = (
+                int(brightness["value"]) if brightness["mode"] == "enabled" else 100
             )
 
-            device_info = await self._client.get_device_info()
+            self._brightness = int(round(brightness_value * 2.55)) if self._is_on else 0
```

Comment:
Just set `https://github.com/home-assistant/core/blob/dev/homeassistant/components/light/__init__.py#L692`, not need to have an property above